qiskit_ibm_runtime/runtime_job.py

```diff
@@ -320,12 +323,42 @@ def _set_error_message(self, job_response: Dict) -> None:
         """
         if self._status == JobStatus.ERROR:
             job_result_raw = self._api_client.job_results(job_id=self.job_id)
-            self._error_message = API_TO_JOB_ERROR_MESSAGE[
-                job_response["status"].upper()
-            ].format(self.job_id, job_result_raw)
+            self._error_message = self._error_msg_from_job_response(
+                job_response
+            ).format(self.job_id, job_result_raw)
         else:
             self._error_message = None
 
+    def _error_msg_from_job_response(self, response: Dict) -> str:
+        """Returns the error message from an API response
+
+        Args:
+            response: Job response from the runtime API.
+
+        Returns:
+            Error message.
+        """
+        status = response["state"]["status"].upper()
+        if status == "CANCELLED" and self._reason == "Ran too long":
```

Comment:
Previously we used a case insensitive check. Do you think this important or can we rely on the API to use this exact capitalization?

Reply:
To be safe, I think we can do a case insensitive check, the capitalization may be inconsistent in the future